config/configuration_singleton.rb

```diff
@@ -80,7 +80,7 @@ def announcement_path
   # The paths to the JSON files that store the quota information
   # @return [Array<Pathname>] quota paths
   def quota_paths
-    ENV.fetch("OOD_QUOTA_PATH", "").strip.split(":").map {|p| Pathname.new(p)}
+    ENV.fetch("OOD_QUOTA_PATH", "").strip.split(/(?<!\\):/).map {|p| p.gsub(/\\/,"")}
```

Comment:
One final thought:

The ability to escape colon in a colon delimited string does enable adding any path or URI that contains the colon, but at the cost of having to escape using the `\\`.

If we add a negative lookahead for `//`, assuming that normal paths would never start with `//`, then we can support http and ftp URIs without escaping:

```ruby
ENV.fetch("OOD_QUOTA_PATH", "").strip.split(/(?<!\\):(?!\/\/)/).map{ |p| p.gsub(/\\/,"") }
```

For example

```
irb(main):008:0> path = "/project/rice/d/ddietz:https://www.rcac.purdue.edu/ws/storagedirquota/ddietz:/scratch/rice/d/ddietz:/path/with/\\:colon/in/it"
=> "/project/rice/d/ddietz:https://www.rcac.purdue.edu/ws/storagedirquota/ddietz:/scratch/rice/d/ddietz:/path/with/\\:colon/in/it"
irb(main):016:0> path.split(/(?<!\\):(?!\/\/)/).map{ |p| p.gsub(/\\/,"") }
=> ["/project/rice/d/ddietz", "https://www.rcac.purdue.edu/ws/storagedirquota/ddietz", "/scratch/rice/d/ddietz", "/path/with/:colon/in/it"]
```

Another option is to drop support for escaping the colon, since the real goal right now is to only add support for URLs. And in that case we can do:

```ruby
def quota_paths
  ENV.fetch("OOD_QUOTA_PATH", "").strip.split(/:(?!\/\/)/)
end
```

then:

```
irb(main):018:0> path = "/project/rice/d/ddietz:https://www.rcac.purdue.edu/ws/storagedirquota/ddietz:/scratch/rice/d/ddietz"
=> "/project/rice/d/ddietz:https://www.rcac.purdue.edu/ws/storagedirquota/ddietz:/scratch/rice/d/ddietz"
irb(main):019:0> path.split(/:(?!\/\/)/)
=> ["/project/rice/d/ddietz", "https://www.rcac.purdue.edu/ws/storagedirquota/ddietz", "/scratch/rice/d/ddietz"]
```

IMHO this would be my preference since the code is a little simpler and we don't have a use case for escaping the colon beyond supporting URLs. @MorganRodgers any thoughts?

Reply:
I prefer the second as it is simpler but still meets our needs. Symbol heavy regex take longer to grok.

Reply:
@ddietz89 I just wanted to follow up and confirm if you were okay with the proposed change, where instead of supporting escaping colon in the OOD_QUOTA_PATH, we instead would use negative lookahead for `//` to ignore any colon that preceeded `//` which would enable support for all URLs of the format `scheme://`

Reply:
@ericfranz Sorry, have been rolling out OOD site wide and forgot about this. Yes, this makes sense. I switched to this method in my deployed version of the code. Thanks!

Reply:
Done! https://github.com/OSC/ood-dashboard/commit/f41ff2ea9f0e9f28b34055d522758f2fec1b4cbb